docs/formats/zjson.md

```diff
@@ -0,0 +1,492 @@
+```
+{
+    ts: 2018-03-24T17:15:21.926018012Z,
+    a: "hello, world",
+    b: {
+        x: 4611686018427387904,
+        y: 127.0.0.1
+    }
+}
+```
+A straightforward translation to JSON might look like this:
+```
+{
+  "ts": 1521911721.926018012,
+  "a": "hello, world",
+  "b": {
+    "x": 4611686018427387904,
+    "y": "127.0.0.1"
+  }
+}
+```
+But, when this JSON is transmitted to a JavaScript client and parsed,
+the result looks something like this:
+```
+{
+  "ts": 1521911721.926018,
+  "a": "hello, world",
+  "b": {
+    "x": 4611686018427388000,
+    "y": "127.0.0.1"
+  }
+}
+```
+The good news is the `a` field came through just fine, but there are
+a few problems with the remaining fields:
+* the timestamp lost precision (due to 53 bits of mantissa in a JavaScript
+IEEE 754 floating point number) and was converted from a time type to a number,
+* the int64 lost precision for the same reason, and
+* the IP address has been converted to a string.
+
+As a comparison, Python's `json` module handles the 64-bit integer to full
+precision, but loses precision on the floating point timestamp.
+Also, as mentioned, it is at the whim of a JSON implementation whether
+or not the order of object keys is preserved.
+
+While JSON is well suited for data exchange of generic information, it is not
+so appropriate for a structured data model like Zed.
+That said, JSON can be used as an encoding format for Zed by mapping Zed data
+onto a JSON-based protocol.  This allows clients like web apps or
+electron apps to receive and understand Zed and, with the help of client
+libraries like [zealot](https://github.com/brimdata/brim/tree/master/zealot),
+to be enabled with rich, structured Zed types that are implemented on top of
+the basic JavaScript types.
+
+In other words,
+because JSON objects do not have a deterministic column order nor does JSON
+in general have typing beyond the basics (i.e., strings, floating point numbers,
+objects, arrays, and booleans), we decided to encode Zed data with
+its embedded type model all in a layer above regular JSON.
+
+## ZJSON
+
+The format for representing Zed in JSON is called ZJSON.
+Converting ZSON/ZNG/ZST to ZJSON and back results in a complete and
+accurate restoration of the original Zed data.
+
+The ZJSON data model follows that of the underlying Zed model by embedding
+type information in the stream: type definitions declare arbitrarily complex
+and nested data types, and values are sent referencing the type information
+recursively with small-integer type identifiers.
+
+Since Zed steams are self describing and type information is embedded
+in the stream itself, the embedded types are likewise encoded in the
+ZJSON format.
+
+A ZJSON stream is defined as a sequence of JSON objects where each object
+represents a Zed value.  Each object includes an identifier that denotes
+its type, or _schema_.  A schema generically refers to the type of the
+Zed record that is defined by a given JSON object.
```

Comment:
This paragraph refers to "a Zed value" and then, later, "the Zed record". I think it will read better sticking to one or the other.

Reply:
Good catch.  ZJSON actually works for the generic Zed model (any stream of values not just records) so I adopted "Zed value" over record.  (Not that we've implemented this... :-)